crates/findex/src/implementations/redis/findex.rs

```diff
@@ -0,0 +1,434 @@
+use std::{
+    collections::{HashMap, HashSet},
+    sync::Arc,
+};
+
+use cosmian_findex::{
+    parameters::{
+        BLOCK_LENGTH, CHAIN_TABLE_WIDTH, KMAC_KEY_LENGTH, KWI_LENGTH, MASTER_KEY_LENGTH, UID_LENGTH,
+    },
+    EncryptedMultiTable, EncryptedTable, FetchChains, FindexCallbacks, FindexCompact, FindexSearch,
+    FindexUpsert, IndexedValue, KeyingMaterial, Keyword, Label, Location, Uid, Uids, UpsertData,
+};
+use redis::{aio::ConnectionManager, pipe, AsyncCommands, Script};
+use tokio::sync::RwLock;
+use tracing::trace;
+
+use super::{error::FindexRedisError, RemovedLocationsFinder};
+
+#[derive(Copy, Clone)]
+enum FindexTable {
+    Entry,
+    Chain,
+}
+
+/// Generate a key for the entry table or chain table
+fn key(table: FindexTable, uid: &[u8]) -> Vec<u8> {
+    let mut key = match table {
+        FindexTable::Entry => b"fe::".to_vec(),
+        FindexTable::Chain => b"fc::".to_vec(),
+    };
+    key.extend_from_slice(uid);
+    key
+}
```

Comment:
Yes, ET and CT are stored in the same Redis (Redis doesn't have the concept of table).

In Redis you need to have a prefix to split the keys for `fetch_all_entry_table_uids` you cannot add a boolean to the value and filter on it. But the prefix could be smaller (I'm not sure it's worth it)

Reply:
I think this is worth modifying:
- 4 characters -> 12.5% overhead on UID storage -> ~4% overhead on the Entry Table and 2/3% overhead on the Chain Table
- the copy of each UID in a newly allocated vector wastes time and memory

I think it may be possible to simply set the first bit of each UID to 1 or 0. No overhead, no extra allocation and the UID size guarantee there will be no collision. I will double-check with @chloehebant.

Reply:
The Redis server can be used to store other things (in the KMS, the KMIP objects for instance), so I would like to keep the prefix discriminant enough but have reduced it to size 2

The code is now

```
/// The length of the prefix of the table name in bytes
/// 0x00ee for the entry table
/// 0x00ef for the chain table
const TABLE_PREFIX_LENGTH: usize = 2;

#[derive(Copy, Clone)]
enum FindexTable {
    Entry = 0xee,
    Chain = 0xef,
}

/// Generate a key for the entry table or chain table
fn key(table: FindexTable, uid: &[u8]) -> Vec<u8> {
    [&[0x00, table as u8], uid].concat()
}
```